apps/payments/next/app/[locale]/subscriptions/landing/route.ts

```diff
@@ -0,0 +1,65 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import { signIn } from 'apps/payments/next/auth';
+import { NextRequest } from 'next/server';
+import { config } from 'apps/payments/next/config';
+import { getMetricsFlowAction } from '@fxa/payments/ui/actions';
+import { ManageParams } from '@fxa/payments/ui';
+import { redirect } from 'next/navigation';
+
+export const dynamic = 'force-dynamic'; // defaults to auto
+
+export async function GET(
+  request: NextRequest,
+  { params }: { params: ManageParams }
+) {
+  const requestSearchParams = request.nextUrl.searchParams;
+  const { locale } = params;
+
+  const redirectToUrl = new URL(
+    `${config.paymentsNextHostedUrl}/${locale}/subscriptions/manage`
+  );
+  redirectToUrl.search = requestSearchParams.toString();
+
+  if (
+    !redirectToUrl.searchParams.has('flow_id') &&
+    !redirectToUrl.searchParams.has('flow_begin_time')
+  ) {
+    try {
+      const metricsFlow = await getMetricsFlowAction();
+      redirectToUrl.searchParams.set('flow_id', metricsFlow.flowId);
+      redirectToUrl.searchParams.set(
+        'flow_begin_time',
+        metricsFlow.flowBeginTime.toString()
+      );
+    } catch (error) {
+      // Prevent error propagation on failed metrics flow fetch
+      console.error(error);
+    }
+  }
+
+  let redirectUrl;
+  try {
+    redirectUrl = await signIn(
+      'fxa',
+      {
+        redirect: true,
+        redirectTo: redirectToUrl.href,
+      },
+      { prompt: 'none' }
+    );
+  } catch (error) {
+    try {
+      redirectUrl = await signIn('fxa', {
+        redirect: false,
+        redirectTo: redirectToUrl.href,
+      });
+    } catch (error) {
+      // Prevent error propagation on failed auth to allow redirect
+      console.error(error);
+    }
+  }
```

Comment:
issue: Oooh, my apologies, I just realized that this approach won't work.

The output of this `/landing` route should be a redirect to the FxA auth page and providing a URL for the FxA auth page to redirect the browser to after the auth attempt.

With that in mind, since the auth attempt only occurs after the request to this `/landing` route has already completed, it's not possible to "catch" the result of the no-prompt login in case of a no-prompt auth failure, and then instead try a regular sign in.

I didn't go super deep into detail, so I'm happy to discuss this more.

Reply:
suggestion: I believe, the Mozilla Accounts, team recently added new functionality for no-prompt, to automatically redirect to the Sign In page on no-prompt auth failure.

https://mozilla-hub.atlassian.net/browse/FXA-11730